### ISSUE TITLE
BulkInsert fails when FK is a shadow property on SQLite

### ISSUE THREAD

**Issue by sandrohanea** [zzzprojects/EntityFramework-Extensions#258]
### Description
I cannot use BulkInsert with EFCore and SQLite when having some related entities with shadow primary key and IncludeGraph = true;

### Exception
SQLite Error 19: 'FOREIGN KEY constraint failed'.

```
Exception message:
Stack trace:
Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 19: 'FOREIGN KEY constraint failed'.
at Microsoft.Data.Sqlite.SqliteException.ThrowExceptionForRC(Int32 rc, sqlite3 db)
at Microsoft.Data.Sqlite.SqliteCommand.ExecuteReader(CommandBehavior behavior)
at Microsoft.Data.Sqlite.SqliteCommand.ExecuteDbDataReader(CommandBehavior behavior)
at System.Data.Common.DbCommand.ExecuteReader()
at .(DbCommand , BulkOperation )
at .( , DbCommand )
at .Execute(List1 actions) at .(List1 )
at Z.BulkOperations.BulkOperation.Execute()
at Z.BulkOperations.BulkOperation.BulkInsert()
at .[](DbContext this, IEnumerable1 , Action1 )
at .(DbContext this, StateManager , IReadOnlyList1 , Action1 )
at .(DbContext this, StateManager , IReadOnlyList1 , Action1 )
at .(DbContext this, Action1 ) at DbContextExtensions.BulkSaveChanges(DbContext this, Action1 options)
at .[](DbContext this, IEnumerable1 , Action1 )
at DbContextExtensions.BulkInsert[T](DbContext this, IEnumerable1 entities, Action1 options)

Code used to bulk insert:

this.dbContext.BulkInsert(entitiesToAdd, options =>
{
options.IncludeGraph = true;
});
I tried to disable primary keys before inserting but the error is still here.
"PRAGMA foreign_keys = 0;";

**Comment by JonathanMagnan** [comment]
Hello @sandrohanea ,

Thank you for reporting,

We will try to reproduce it today. If we don't succeed, we will ask you for a runnable project.

(Ref: https://github.com/zzzprojects/EntityFramework-Extensions/issues/94)

Best Regards,

Jonathan

**Comment by JonathanMagnan** [comment]
Hello @sandrohanea ,

The v2.6.11 has been released.

Could you try this version? My developer fixed an issue similar to your.

Let me know if we successfully fixed it or you still get the error.

**Comment by sandrohanea** [comment]
Hi, I updated it and the issue seems solved. Still, I had another issue with OwnsOne method.

I uploaded a demo project here:
https://github.com/sandrohanea/DemoEfCoreBulkInsert

I managed to work around this issue by replacing OwnsOne with HasOne

**Comment by sandrohanea** [comment]
Thank you for your rapid fix :) 
Have a nice day!

**Comment by JonathanMagnan** [comment]
Awesome ;)

We will look at your second issue.

**Comment by sandrohanea** [comment]
Hi again!
I found the third issue:
After inserting data with BulkInsert, if the entity has a GUID primary key, then, we cannot retrieve these entities using something like 
`dbContext.EntitiesGuid.Include(e` => e.ChildList).FirstOrDefault(x => x.Id == anId)`
But the entities have the GUID correctly set.
I attached a repo with the repro of this issue in https://github.com/sandrohanea/DemoEfCoreBulkInsert/tree/GuidPrimaryKeyError.

Should I create a new issue with this?

Thanks in advance!
Have a nice day!

**Comment by JonathanMagnan** [comment]
That's fine @sandrohanea ,

We can continue on this thread. My developer will be able to check both issues tomorrow.

**Comment by JonathanMagnan** [comment]
Hello @sandrohanea ,

Today, we have released the v2.6.12-beta1,

We currently don't have time to end this request this week (vacancy) but I believe already everything is here and should be fixed for your both issue.

Next week, we will complete the code review/test to release a stable version but if you could meanwhile try it to see if we forget something, that would be great.

Best Regards,

Jonathan

**Comment by sandrohanea** [comment]
I checked both of the issues(OwnsOne, which is not very critical as there's a workaround for it) and Guid Primary Key and none of them was fixed in 7.6.12-beta1 :(

Best Regards,

Sandro

**Comment by JonathanMagnan** [comment]
That's weird, we will look at it.
